src/main/java/net/helpscout/api/ApiClient.java

```diff
@@ -1452,61 +1450,46 @@ private void setThreadProperties(ConversationThread thread) {
     }
     
     private <T> T doPost(String url, String requestBody, int expectedCode, ResultExtractor<T> extractor) throws ApiException {
-        HttpURLConnection conn = null;
-        try {
-            conn = HTTPConnectionUtils.getConnection(apiKey, baseUrl + url, METHOD_POST, expectedCode, requestBody);
+
+        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_POST, expectedCode, requestBody)) {
             return extractor.extract(conn);
         } catch(ApiException ex) {
             throw ex;
         } catch (Exception ex) {
             throw new RuntimeException(ex);
-        } finally {
-            HTTPConnectionUtils.close(conn);
         }
     }
 
     private void doPut(String url, String requestBody, int expectedCode) throws ApiException {
-        HttpURLConnection conn = null;
-        try {
-            conn = HTTPConnectionUtils.getConnection(apiKey, baseUrl + url, METHOD_PUT, expectedCode, requestBody);
+
+        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_PUT, expectedCode, requestBody)) {
         } catch(ApiException ex) {
             throw ex;
         } catch (Exception ex) {
             ex.printStackTrace();
             throw new RuntimeException(ex);
-        } finally {
-            HTTPConnectionUtils.close(conn);
         }
     }
 
     private String doGet(String url, int expectedCode) throws ApiException {
-        HttpURLConnection conn = null;
         String response    = null;
-        try {
-            conn = HTTPConnectionUtils.getConnection(apiKey, baseUrl + url, METHOD_GET, expectedCode);
-            response = HTTPConnectionUtils.getResponse(conn);
-
+        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_GET, expectedCode)) {
```

Comment:
@bob983 as architectural it good, I agree, but I have a little bit different point of view. I consider it no need to make something more complex that it is until it is necessary. As for I do not see what kind of other connection could appear in API in future and prepared flexibility will be utilised, so I suppose it should be treated as a usual class without an abstraction layer. What do you think? Do you have in plan to add some other connectionWrapper? Do not get me wrong, it is really easy to add for me:) I just like to understand. :)